lib/puppet/provider/iis_powershell.rb

```diff
@@ -57,6 +58,50 @@ def self.ps_manager
     PuppetX::IIS::PowerShellManager.instance("#{command(:powershell)} #{PuppetX::IIS::PowerShellCommon.powershell_args.join(' ')}")
   end
 
+  def self.ps_major_version
+    if @ps_major_version.nil?
+      version = PuppetX::IIS::PowerShellVersion.version
+      @ps_major_version = version.split('.').first.to_i
+    end
+    @ps_major_version
+  end
+
+  def self.add_powershell2_support
+    # - PowerShell 2.0 does not support autoload of modules therefore we must explicitly add the WebAdministration module
+    # - Must change the current location to the be the IIS: provider
```

Comment:
This seems to be because we are running PowerShell in a custom host instead of interactively.  Without the CD: Get-WebSite fails with a Null Exception.

Other examples:
https://forums.iis.net/t/1188540.aspx?Get+Website+Object+reference+not+set+to+an+instance+of+an+object+

https://forums.iis.net/t/1222290.aspx?Bug+report+NullReferenceException+when+using+the+WebAdministration+cmdlets+in+a+workflow